ouroboros-network/src/Ouroboros/Network/StrictFingerTree.hs

```diff
@@ -0,0 +1,264 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+-- | Strict variants of 'FingerTree' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.FingerTree.Strict@.
+module Ouroboros.Network.StrictFingerTree
+  ( FingerTree
+
+    -- * Construction
+  , empty
+  , singleton
+  , (<|)
+  , (|>)
+  , (><)
+  , fromList
+
+    -- * Deconstruction
+  , null
+
+    -- ** Examining the ends
+  , ViewL(..)
+  , ViewR(..)
+  , viewl
+  , viewr
+
+    -- ** Search
+  , SearchResult(..)
+  , search
+
+    -- ** Splitting
+    -- | These functions are special cases of 'search'.
+  , split
+  , takeUntil
+  , dropUntil
+
+    -- * Transformation
+  , reverse
+
+    -- ** Maps
+  , fmap'
+
+  -- Re-export from "Data.FingerTree"
+  , Measured(..)
+  ) where
+
+import           Prelude hiding (null, reverse)
+
+import           Data.FingerTree (Measured (..))
+import qualified Data.FingerTree as FT
+import           Data.Foldable (foldl')
+import           GHC.Generics (Generic)
+
+infixr 5 ><
+infixr 5 <|, :<
+infixl 5 |>, :>
+
+-- | A @newtype@ wrapper around a 'FT.FingerTree', representing a finger tree
+-- that is strict in its values.
+--
+-- This strictness is not enforced at the type level, but rather by the
+-- construction functions in this module. These functions essentially just
+-- wrap the original "Data.FingerTree" functions while forcing the provided
+-- value to WHNF.
+newtype FingerTree v a = SFT (FT.FingerTree v a)
```

Comment:
The type should be called `StrictFingerTree`; Warning and error messages will be easier to digest.